### PR TITLE
Fix the daily pipeline failures

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

1. Fix the nuget cpu pipeline and put code coverage pipeline back. 
2. Reduce onnx_test_runner's default logging level from WARNING to ERROR. Because there are too many log messages now. 
3. Enlarge the protobuf read buffer size for onnx_test_runner. It was missed from PR #4020.
**Motivation and Context**
- Why is this change required? What problem does it solve?

Currently, 
1. The nuget CPU pipeline fails with error of:
"curl: command not found" 
while downloading azcopy which is for downloading the test data. But because I've put azcopy and all the test on the machines, so there is no need to re-download them from Azure again. To fix the problem, instead of installing curl on the machines, I deleted the data downloading step. If later on we need to add the step back, please only bring the "Download test data" step back, not the "Download azcopy". Because the azcopy command is already available at "/usr/bin/".

2. The code coverage pipeline fails because it uses latest protobuf release and our code isn't compatible with that. I've sent the issue to Pranav. Before we get a fix,  I want to migrate the pipeline to Linux and LLVM. The LLVM code coverage tool has better documentation and it is easier to configure. It can merge the coverage result from multiple runs without extra tools. 

- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker-gpu.sh [hunk 1]
```diff
@@ -13,44 +13,28 @@ CurrentOnnxRuntimeVersion=$4
...
 IMAGE="ubuntu16.04-$CUDA_VER"
 OldDir=$(pwd)
 
 cd $SOURCE_ROOT/tools/ci_build/github/linux/docker
 
-DOCKER_FILE=Dockerfile.ubuntu_gpu_cuda9
-if [ $CUDA_VER = "cuda10.0-cudnn7.3" ]; then
-DOCKER_FILE=Dockerfile.ubuntu_gpu
-fi
+sudo --preserve-env docker build -t onnxruntime-ubuntu16.04-cuda10.1-cudnn7.6 --build-arg OS_VERSION=16.04 --build-arg BUILD_USER=onnxruntimedev --build-arg BUILD_UID=$(id -u) --build-arg PYTHON_VERSION=${PYTHON_VER}  -f Dockerfile.ubuntu_gpu .
```
Comment (yuslepukhin) [inline csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker-gpu.sh:23]: >--preserve-env [](start = 5, length = 14)

Was not env preserved before?

File: onnxruntime/test/onnx/TestCase.cc [hunk 2]
```diff
@@ -27,6 +27,8 @@ using namespace onnxruntime;
 using namespace onnxruntime::common;
 using google::protobuf::RepeatedPtrField;
 
+static constexpr int protobuf_block_size = 1 << 20;
```
Comment (yuslepukhin) [inline onnxruntime/test/onnx/TestCase.cc:30]: >protobuf_block_size [](start = 21, length = 19)

In KByes?

File: tools/ci_build/github/azure-pipelines/nuget/templates/gpu.yml [hunk 3]
```diff
@@ -305,4 +306,4 @@ jobs:
     inputs:
       command: 'push'
       packagesToPush: '$(Build.BinariesDirectory)/nuget-artifact/final-package/Microsoft.ML.OnnxRuntime.DirectML.$(DirectMLNuGetPackageVersionNumber).nupkg'
-      publishVstsFeed: '2692857e-05ef-43b4-ba9c-ccf1c22c437c/7982ae20-ed19-4a35-a362-a96ac99897b7'
\ No newline at end of file
+      publishVstsFeed: '2692857e-05ef-43b4-ba9c-ccf1c22c437c/7982ae20-ed19-4a35-a362-a96ac99897b7'
```
Comment (yuslepukhin) [inline tools/ci_build/github/azure-pipelines/nuget/templates/gpu.yml:309]: >ac99897b7' [](start = 88, length = 10)

Is it a whitespace change?

File: tools/ci_build/github/azure-pipelines/post-merge-jobs.yml [hunk 4]
```diff
@@ -1,107 +1,34 @@
...
+          llvm-profdata merge -sparse  -o ort.profdata *.profraw
+          llvm-cov export -summary-only -instr-profile=ort.profdata onnx_test_runner -object onnxruntime_test_all  $(Build.SourcesDirectory)/include/onnxruntime $(Build.SourcesDirectory)/onnxruntime/core $(Build.SourcesDirectory)/onnxruntime/contrib_ops > $(Build.BinariesDirectory)/report.json
+        workingDirectory: $(Build.BinariesDirectory)
+
+
+    - task: CmdLine@2
+      displayName: 'Post code coverage data to the Dashboard'
+      inputs:
+        script: |
+          python3 $(Build.SourcesDirectory)/tools/ci_build/github/windows/post_code_coverage_to_dashboard.py  --commit_hash=$(OnnxRuntimeGitCommitHash) --report_file="$(Build.BinariesDirectory)/report.json" --report_url="https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=$(Build.BuildId)"
```
Comment (yuslepukhin) [inline tools/ci_build/github/azure-pipelines/post-merge-jobs.yml:29]: >"https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=$(Build.BuildId)" [](start = 220, length = 87)

Is this consumed by a dahsboard? Can we put more comments into files?

**Comment by yuslepukhin** [comment]
>   displayName: 'Clean untagged docker images'

What kind of vars this script sets and what does it download? Should we rename this script? Perhaps some comments in the header?

---
Refers to: tools/ci_build/github/azure-pipelines/nuget/templates/linux-set-variables-and-download.yml:3 in 79c19b8. [](commit_id = 79c19b82d2b5c68b567397f86af2bd8d575546dd, deletion_comment = False)

File: tools/ci_build/github/azure-pipelines/nuget/templates/test_win.yml [hunk 5]
```diff
@@ -62,6 +59,11 @@ jobs:
     parameters:
       packageFolder: '$(Build.BinariesDirectory)\nuget-artifact'
 
+  - script: |
+     mklink /D /J models C:\local\models
```
Comment (yuslepukhin) [inline tools/ci_build/github/azure-pipelines/nuget/templates/test_win.yml:63]: >      [](start = 40, length = 5)

Trailing spaces

**Comment by yuslepukhin** [comment]
Would be nice to describe changes here on a high level. Fix Nuget pipeline is too general What specifically needs to be fixed.
This comments will lead to less questions being asked in the future.

File: onnxruntime/test/onnx/TestCase.cc (hunk 2 above)
Comment (snnn) [reply onnxruntime/test/onnx/TestCase.cc:30]: Bytes. 1<<20 is 1MB.

Comment (snnn) [reply onnxruntime/test/onnx/TestCase.cc:30]: See #4020

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker-gpu.sh (hunk 1 above)
Comment (snnn) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker-gpu.sh:23]: I guess it's not. See the changes in #4063

File: tools/ci_build/github/azure-pipelines/nuget/templates/gpu.yml (hunk 3 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/nuget/templates/gpu.yml:309]: Yes.

File: tools/ci_build/github/azure-pipelines/post-merge-jobs.yml (hunk 4 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/post-merge-jobs.yml:29]: Yes. I'll will.

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker-gpu.sh (hunk 1 above)
Comment (edgchen1) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker-gpu.sh:23]: yep. `env` and `sudo env` may be quite different

File: onnxruntime/test/onnx/main.cc [hunk 6]
```diff
@@ -224,9 +224,9 @@ int real_main(int argc, char* argv[], Ort::Env& env) {
 
   // set log level based on number of verbosity options
   if (verbosity_option_count == 1) {
-    logging_level = ORT_LOGGING_LEVEL_INFO;
+    logging_level = ORT_LOGGING_LEVEL_WARNING;
   } else if (verbosity_option_count > 1) {
-    logging_level = ORT_LOGGING_LEVEL_VERBOSE;
+    logging_level = ORT_LOGGING_LEVEL_INFO;
```
Comment (edgchen1) [inline onnxruntime/test/onnx/main.cc:229]: how about something like this for more flexibility:
`logging_level = ORT_LOGGING_LEVEL_ERROR - std::min(verbosity_option_count, ORT_LOGGING_LEVEL_ERROR)`
also i think verbosity_option_count can be unsigned

File: onnxruntime/test/onnx/TestCase.cc (hunk 2 above)
Comment (edgchen1) [reply onnxruntime/test/onnx/TestCase.cc:30]: how about `protobuf_block_size_in_bytes`?

File: onnxruntime/test/onnx/main.cc (hunk 6 above)
Comment (snnn) [reply onnxruntime/test/onnx/main.cc]: Please check.

File: tools/ci_build/github/azure-pipelines/nuget/templates/gpu.yml (hunk 3 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/nuget/templates/gpu.yml]: Reverted.

File: tools/ci_build/github/windows/post_code_coverage_to_dashboard.py [hunk 7]
```diff
@@ -10,27 +10,30 @@
 
 import argparse
```
Comment (edgchen1) [inline tools/ci_build/github/windows/post_code_coverage_to_dashboard.py:11]: looks like this is called from Linux now, maybe tools/ci_build/github/windows is not the best location. can address it in a separate PR

File: onnxruntime/test/onnx/main.cc (hunk 6 above)
Comment (edgchen1) [reply onnxruntime/test/onnx/main.cc]: looks good. i think we can still make verbosity_option_count unsigned to ensure it won't be negative, but it currently looks like it shouldn't be

**Comment by snnn** [comment]
It's known that the nuget GPU pipeline will continue to fail. I feel sorry for that, I would like to let this change go first and put the fix in another PR if nobody disagree.

**Pull request merged**